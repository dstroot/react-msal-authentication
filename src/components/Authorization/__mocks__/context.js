// create a test user
const user = {
  displayableId: "dan@msft.com",
  name: "User Name",
  identityProvider: "https://login.microsoftonline.com/v2.0",
  userIdentifier: "identfierstring",
  idToken: {
    aud: "uniquestring",
    iss: "https://login.microsoftonline.com/v2.0",
    iat: 1549125156,
    nbf: 1549125156,
    exp: 1549129056,
    aio: "anotheruniquestring",
    name: "User Name",
    nonce: "string",
    oid: "string",
    preferred_username: "dan@msft.com",
    sub: "string",
    tid: "string",
    uti: "striung",
    ver: "2.0"
  }
};

// create mock context data to use for our test
const context = {
  authenticated: true,
  wait: false,
  user: user,
  login: jest.fn(),
  logout: jest.fn()
};

export default context;
