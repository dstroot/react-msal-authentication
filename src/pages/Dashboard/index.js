import React from "react";
import { AuthConsumer } from "../../components/AuthContext";
// import "./styles.scss";

const Dashboard = () => (
  <AuthConsumer>
    {({ user }) => {
      // let minutes = 1000 * 60;
      // let now = new Date().getTime();
      // let expires = user.idToken.exp * 1000;
      // let timeDiff = expires - now;
      // let time = Math.floor(timeDiff / minutes);
      return (
        <div className="container mt-5">
          <div className="row">
            <div className="col">
              <h2>User Dashboard</h2>
              <h3>
                User: {user.name} ({user.displayableId})
              </h3>
              {/*<h5>
                Authenticated: {(now < expires).toString()} (expires: {time}{" "}
                minutes)
              </h5>*/}
              <h5>User Object</h5>
              <pre>{JSON.stringify(user, null, 4)}</pre>
            </div>
          </div>
        </div>
      );
    }}
  </AuthConsumer>
);

export default Dashboard;
