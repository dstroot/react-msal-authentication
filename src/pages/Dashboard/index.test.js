import React from "react";
import renderer from "react-test-renderer";

// import component to test
import Dashboard from "../Dashboard";
import context from "../../components/Authorization/__mocks__/context";

// Mock our AuthConsumer
const mockContext = jest.fn();
jest.mock("../../components/Authorization", () => ({
  AuthConsumer: ({ children }) => children(mockContext())
}));

// Test Dashboard page
describe("Dashboard", () => {
  // mount the component
  let mountedComponent;
  const getMountedComponent = () => {
    if (!mountedComponent) {
      mountedComponent = mount(<Dashboard />);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
    mockContext.mockReset();
  });

  it(`should render`, () => {
    // set our context data
    mockContext.mockReturnValue(context);

    // test rendering
    const component = renderer.create(<Dashboard />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should pass context", () => {
    mockContext.mockReturnValue(context);
    const pre = getMountedComponent()
      .find("pre")
      .first();

    // see if the "pre" tag contains context data
    expect(pre.text()).toContain(
      `"identityProvider": "https://login.microsoftonline.com/v2.0"`
    );
  });
});
