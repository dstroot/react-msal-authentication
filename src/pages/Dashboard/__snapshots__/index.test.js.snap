// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render 1`] = `
<div
  className="container mt-5"
>
  <div
    className="row"
  >
    <div
      className="col"
    >
      <h2>
        User Dashboard
      </h2>
      <h3>
        User: 
        User Name
         (
        dan@msft.com
        )
      </h3>
      <h5>
        User Object
      </h5>
      <pre>
        {
    "displayableId": "dan@msft.com",
    "name": "User Name",
    "identityProvider": "https://login.microsoftonline.com/v2.0",
    "userIdentifier": "identfierstring",
    "idToken": {
        "aud": "uniquestring",
        "iss": "https://login.microsoftonline.com/v2.0",
        "iat": 1549125156,
        "nbf": 1549125156,
        "exp": 1549129056,
        "aio": "anotheruniquestring",
        "name": "User Name",
        "nonce": "string",
        "oid": "string",
        "preferred_username": "dan@msft.com",
        "sub": "string",
        "tid": "string",
        "uti": "striung",
        "ver": "2.0"
    }
}
      </pre>
    </div>
  </div>
</div>
`;
